opm/grid/cpgrid/CpGridData.hpp

```diff
@@ -257,6 +285,592 @@ class CpGridData
+            const std::array<int,3>& grid_dim = this -> logicalCartesianSize();
+            // Select 8 corners of the patch boundary to be the 8 corners of the 'cellified patch'.
+            cellifiedPatch_to_point = { // Corner-index: (J*(grid_dim[0]+1)*(grid_dim[2]+1)) + (I*(grid_dim[2]+1)) +K
+                // Index of corner '0' {startI, startJ, startK}
+                (startIJK[1]*(grid_dim[0]+1)*(grid_dim[2]+1)) + (startIJK[0]*(grid_dim[2]+1)) + startIJK[2],
+                // Index of corner '1' '{endI, startJ, startK}'
+                (startIJK[1]*(grid_dim[0]+1)*(grid_dim[2]+1)) + (endIJK[0]*(grid_dim[2]+1)) + startIJK[2],
+                // Index of corner '2' '{startI, endJ, startK}'
+                (endIJK[1]*(grid_dim[0]+1)*(grid_dim[2]+1)) + (startIJK[0]*(grid_dim[2]+1)) + startIJK[2],
+                // Index of corner '3' '{endI, endJ, startK}'
+                (endIJK[1]*(grid_dim[0]+1)*(grid_dim[2]+1)) + (endIJK[0]*(grid_dim[2]+1)) + startIJK[2],
+                // Index of corner '4' '{startI, startJ, endK}'
+                (startIJK[1]*(grid_dim[0]+1)*(grid_dim[2]+1)) + (startIJK[0]*(grid_dim[2]+1))+ endIJK[2],
+                // Index of corner '5' '{endI, startJ, endK}'
+                (startIJK[1]*(grid_dim[0]+1)*(grid_dim[2]+1)) + (endIJK[0]*(grid_dim[2]+1)) + endIJK[2],
+                // Index of corner '6' '{startI, endJ, endK}'
+                (endIJK[1]*(grid_dim[0]+1)*(grid_dim[2]+1)) + (startIJK[0]*(grid_dim[2]+1)) + endIJK[2],
+                // Index of corner '7' {endI, endJ, endK}
+                (endIJK[1]*(grid_dim[0]+1)*(grid_dim[2]+1)) + (endIJK[0]*(grid_dim[2]+1)) + endIJK[2]};
+            EntityVariableBase<cpgrid::Geometry<0,3>>& cellifiedPatch_corners =
+                cellifiedPatch_geometry.geomVector(std::integral_constant<int,3>());
+            cellifiedPatch_corners.resize(8);
+            // Compute the center of the 'cellified patch' and its corners.
+            Geometry<0,3>::GlobalCoordinate cellifiedPatch_center = {0., 0.,0.};
+            for (int corn = 0; corn < 8; ++corn) {
+                cellifiedPatch_center +=
+                    (this -> geometry_.geomVector(std::integral_constant<int,3>()).get(cellifiedPatch_to_point[corn]).center())/8.;
+                cellifiedPatch_corners[corn] =
+                    this -> geometry_.geomVector(std::integral_constant<int,3>()).get(cellifiedPatch_to_point[corn]);
+            }
+            // Compute the volume of the 'cellified patch'.
+            double cellifiedPatch_volume = 0.;
+            for (auto& idx : patch_cells) {
+                cellifiedPatch_volume += (this -> geometry_.geomVector(std::integral_constant<int,0>())
+                                         [EntityRep<0>(idx, true)]).volume();
+            }
+            // Indices of 'all the corners', in this case, 0-7 (required to construct a Geometry<3,3> object). 
+            const std::array<int,8>& allcorners_cellifiedPatch = {0,1,2,3,4,5,6,7};
```

Comment:
I am wondering why valgrind or our tests do not complain. This is a local variable, but we are using it for the returned Geometry via `cellifiedPatch_indices_storage_ptr`.

Reply:
Mmm. I see. Possible fix: pass it as argument with & ?